doc: update PROJECT_PROGRESSION.md (#321)
openjs-foundation/cross-project-council: NEW_PROJECT_APPLICATION.md

* Who are the members of the leadership team?
* How are members of the leadership team nominated?
* How are individuals outside of leadership given commit access?

> * Is the project open/willing to move to more open goverance (if not already in palce)?

Reviewer: ```suggestion
* Is the project open/willing to move to more open goverance (if not already in place)?
```

> * Is the project open/willing to move to more open goverance (if not already in place)?

Reviewer 2: I find this phrase "more open governance" to be ambiguous. I suggest we remove the "more" and it is more direct, especially with the parenthetical statement that follows.

Reviewer 3: I would also suggest that we drop the parenthetical and move that to a top level assertion:

```suggestion
* Is the project open/willing to move to open governance, if not already in place?
```

Also, there's a tiny typo fixed in the suggestion: `goverance` => `governance`

> * Is the project willing to adopt the OpenJS Code of Conduct (CoC)?

Reviewer 4: Should these two changes be broken out into a separate update? They are reasonable requests, but they are someone tangential to the bulk of this PR which is about roles for proposal process

Author: @Myles I can, but it is worth it?

Reviewer 4: maybe not if we have consensus.

Reviewer 5: ```suggestion
* Is the project willing to adopt the OpenJS Foundation Code of Conduct (CoC)?
```
